File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/TextDocumentEditParams.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

data class TextDocumentEditParams(
  val uri: String,
  val edits: List<TextEdit>,
  val options: OptionsParams? = null,
)

File: agent/bindings/csharp/TextDocumentEditParams.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class TextDocumentEditParams
  {
    [JsonProperty(PropertyName = "uri")]
    public string Uri { get; set; }
    [JsonProperty(PropertyName = "edits")]
    public TextEdit[] Edits { get; set; }
    [JsonProperty(PropertyName = "options")]
    public OptionsParams Options { get; set; }
  }
}
